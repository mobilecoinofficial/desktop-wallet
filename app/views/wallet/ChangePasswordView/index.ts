import ChangePasswordView from './ChangePasswordView';

export default ChangePasswordView;
