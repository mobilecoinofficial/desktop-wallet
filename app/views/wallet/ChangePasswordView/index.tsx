import React from 'react';
import type { FC } from 'react';

import {
  Box,
  Breadcrumbs,
  Container,
  FormHelperText,
  FormLabel,
  Link,
  Typography,
  makeStyles,
} from '@material-ui/core';
import { Formik, Form, Field } from 'formik';
import { TextField } from 'formik-material-ui';
import { useSnackbar } from 'notistack';
import { Link as RouterLink } from 'react-router-dom';
import * as Yup from 'yup';

import { SubmitButton } from '../../../components';
import routePaths from '../../../constants/routePaths';
import useFullService from '../../../hooks/useFullService';
import useIsMountedRef from '../../../hooks/useIsMountedRef';
import type { Theme } from '../../../theme';

const useStyles = makeStyles((theme: Theme) => {
  return {
    button: {
      width: 200,
    },
    cardContainer: {
      paddingBottom: 64,
      paddingTop: 8 * 4,
    },
    cardContent: {
      display: 'flex',
      flexDirection: 'column',
      minHeight: 400,
      padding: theme.spacing(4),
    },
    center: {
      display: 'flex',
      justifyContent: 'center',
    },
    code: {
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'column',
      letterSpacing: '.70rem',
      marginRight: '-.70rem',
      padding: theme.spacing(1),
    },
    form: {
      paddingBottom: theme.spacing(2),
    },
    label: {
      width: '100%',
    },
    modal: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
    },
    paper: {
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    root: {
      backgroundColor: theme.palette.background.dark,
      display: 'flex',
      flexDirection: 'column',
      minHeight: '100vh',
    },
  };
});

const ChangePasswordView: FC = () => {
  const classes = useStyles();
  const { enqueueSnackbar } = useSnackbar();
  const isMountedRef = useIsMountedRef();
  const { changePassword } = useFullService();

  return (
    <Container className={classes.cardContainer} maxWidth="sm">
      <Breadcrumbs separator=">" aria-label="breadcrumb">
        <Link
          color="inherit"
          to={routePaths.APP_SETTINGS}
          component={RouterLink}
        >
          <Typography color="textSecondary">Settings</Typography>
        </Link>
        <Typography color="textPrimary">Change Password</Typography>
      </Breadcrumbs>
      <Box
        alignItems="center"
        display="flex"
        justifyContent="space-between"
        my={3}
        flexDirection="column"
      >
        <Typography variant="body2" color="textSecondary">
          Changing your password will change how we encrypt your secret Entropy.
          You should always use strong (unguessable) passwords and change them
          regularly.
        </Typography>
        <br />
        <Typography variant="body2" color="textSecondary">
          To change your current password, please enter your current password
          and your new password.
        </Typography>
      </Box>
      <Box flexGrow={1} mt={3}>
        <Formik
          initialValues={{
            newPassword: '',
            newPasswordConfirmation: '',
            oldPassword: '',
            submit: null,
          }}
          validationSchema={Yup.object().shape({
            newPassword: Yup.string()
              .min(8, 'New Password must be at least 8 characters in length.')
              .max(99, 'Passwords cannot be more than 99 characters.')
              .required('New Password is required'),
            newPasswordConfirmation: Yup.string()
              .oneOf([Yup.ref('newPassword')], 'Must match New Password')
              .required('New Password Confirmation is required'),
          })}
          isInitialValid={false}
          onSubmit={async (
            values,
            {
              setErrors, setStatus, setSubmitting, resetForm,
            },
          ) => {
            try {
              setSubmitting(true);
              await changePassword(values.oldPassword, values.newPassword);
              if (isMountedRef.current) {
                enqueueSnackbar('Successfully change password!', {
                  variant: 'success',
                });
                setStatus({ success: true });
                setSubmitting(false);
                resetForm();
              }
            } catch (err) {
              if (isMountedRef.current) {
                setStatus({ success: false });
                setErrors({ submit: err.message });
                setSubmitting(false);
              }
            }
          }}
        >
          {({
            errors, isSubmitting, isValid, submitForm,
          }) => {
            return (
              <Form>
                <Box pt={4}>
                  <FormLabel component="legend">
                    <Typography color="primary">Change Password</Typography>
                  </FormLabel>
                  <Field
                    component={TextField}
                    fullWidth
                    label="Old Password"
                    margin="normal"
                    name="oldPassword"
                    type="password"
                  />
                  <Field
                    component={TextField}
                    fullWidth
                    label="New Password"
                    margin="normal"
                    name="newPassword"
                    type="password"
                  />
                  <Field
                    component={TextField}
                    fullWidth
                    label="Confirm New Password"
                    margin="normal"
                    name="newPasswordConfirmation"
                    type="password"
                  />
                </Box>
                {errors.submit && (
                  <Box mt={3}>
                    <FormHelperText error>{errors.submit}</FormHelperText>
                  </Box>
                )}
                <SubmitButton
                  disabled={!isValid || isSubmitting}
                  onClick={submitForm}
                  isSubmitting={isSubmitting}
                >
                  Change Password
                </SubmitButton>
              </Form>
            );
          }}
        </Formik>
      </Box>
    </Container>
  );
};

export default ChangePasswordView;
