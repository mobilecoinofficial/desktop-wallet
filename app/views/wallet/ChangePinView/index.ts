import ChangePinView from './ChangePinView';

export default ChangePinView;
