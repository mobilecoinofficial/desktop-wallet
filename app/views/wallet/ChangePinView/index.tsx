import React from 'react';
import type { FC } from 'react';

import {
  Box,
  Breadcrumbs,
  Container,
  FormHelperText,
  FormLabel,
  Link,
  Typography,
  makeStyles,
} from '@material-ui/core';
import { Formik, Form, Field } from 'formik';
import { TextField } from 'formik-material-ui';
import { useSnackbar } from 'notistack';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink } from 'react-router-dom';
import * as Yup from 'yup';

import { SubmitButton } from '../../../components';
import routePaths from '../../../constants/routePaths';
import useIsMountedRef from '../../../hooks/useIsMountedRef';
import type { Theme } from '../../../theme';
import LocalStore from '../../../utils/LocalStore';
import { makeHash } from '../../../utils/hashing';

const useStyles = makeStyles((theme: Theme) => {
  return {
    button: {
      width: 200,
    },
    cardContainer: {
      paddingBottom: 64,
      paddingTop: 8 * 4,
    },
    cardContent: {
      display: 'flex',
      flexDirection: 'column',
      minHeight: 400,
      padding: theme.spacing(4),
    },
    center: {
      display: 'flex',
      justifyContent: 'center',
    },
    code: {
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'column',
      letterSpacing: '.70rem',
      marginRight: '-.70rem',
      padding: theme.spacing(1),
    },
    form: {
      paddingBottom: theme.spacing(2),
    },
    label: {
      width: '100%',
    },
    modal: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
    },
    paper: {
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    root: {
      backgroundColor: theme.palette.background.dark,
      display: 'flex',
      flexDirection: 'column',
      minHeight: '100vh',
    },
  };
});

const ChangePinView: FC = () => {
  const classes = useStyles();
  const { enqueueSnackbar } = useSnackbar();
  const isMountedRef = useIsMountedRef();
  const { t } = useTranslation('ChangePinView');

  return (
    <Container className={classes.cardContainer} maxWidth="sm">
      <Breadcrumbs separator=">" aria-label="breadcrumb">
        <Link color="inherit" to={routePaths.APP_SETTINGS} component={RouterLink}>
          <Typography color="textSecondary">{t('settingsBreadcrumb')}</Typography>
        </Link>
        <Typography color="textPrimary">{t('changePinBreadcrumb')}</Typography>
      </Breadcrumbs>
      <Box
        alignItems="center"
        display="flex"
        justifyContent="space-between"
        my={3}
        flexDirection="column"
      >
        <Typography variant="body2" color="textSecondary">
          {t('description')}
        </Typography>
        <br />
        <Typography variant="body2" color="textSecondary">
          {t('instructions')}
        </Typography>
      </Box>
      <Box flexGrow={1} mt={3}>
        <Formik
          initialValues={{
            newPin: '',
            newPinConfirmation: '',
            submit: null,
          }}
          validationSchema={Yup.object().shape({
            newPin: Yup.string()
              .min(6, t('PinMin'))
              .max(12, t('PinMax'))
              .required(t('PinRequired')),
            newPinConfirmation: Yup.string()
              .oneOf([Yup.ref('newPin')], t('PinConfirmationRef'))
              .required(t('PinConfirmationRequired')),
          })}
          onSubmit={(values, { setStatus, resetForm }) => {
            const hashedPin = makeHash(values.newPin);
            const localStore = new LocalStore();
            localStore.setHashedPin(String(hashedPin));
            if (isMountedRef.current) {
              enqueueSnackbar(t('enqueue'), {
                variant: 'success',
              });
            }
            setStatus({ success: true });
            resetForm();
          }}
        >
          {({ errors, isSubmitting, dirty, isValid, submitForm }) => {
            return (
              <Form>
                <Box pt={4}>
                  <FormLabel component="legend">
                    <Typography color="primary">{t('formLabel')}</Typography>
                  </FormLabel>
                  <Field
                    component={TextField}
                    fullWidth
                    label={t('newPinLabel')}
                    margin="normal"
                    name="newPin"
                    type="Pin"
                  />
                  <Field
                    component={TextField}
                    fullWidth
                    label={t('PinConfirmationLabel')}
                    margin="normal"
                    name="newPinConfirmation"
                    type="Pin"
                  />
                </Box>
                {errors.submit && (
                  <Box mt={3}>
                    <FormHelperText error>{errors.submit}</FormHelperText>
                  </Box>
                )}
                <SubmitButton
                  disabled={!dirty || !isValid || isSubmitting}
                  onClick={submitForm}
                  isSubmitting={isSubmitting}
                >
                  {t('changePinButton')}
                </SubmitButton>
              </Form>
            );
          }}
        </Formik>
      </Box>
    </Container>
  );
};

export default ChangePinView;
