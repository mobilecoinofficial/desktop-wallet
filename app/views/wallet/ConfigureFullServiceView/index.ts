import ConfigureFullServiceView from './ConfigureFullServiceView';

export default ConfigureFullServiceView;
