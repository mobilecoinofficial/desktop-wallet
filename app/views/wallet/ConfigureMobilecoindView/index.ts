import ConfigureMobilecoindView from './ConfigureMobilecoindView';

export default ConfigureMobilecoindView;
