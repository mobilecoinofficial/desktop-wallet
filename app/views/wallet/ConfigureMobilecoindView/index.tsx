import React from 'react';
import type { FC } from 'react';

import {
  Box,
  Breadcrumbs,
  Container,
  Link,
  Typography,
  makeStyles,
} from '@material-ui/core';
import { Link as RouterLink } from 'react-router-dom';

import routePaths from '../../../constants/routePaths';
import type { Theme } from '../../../theme';
import LeaveMobilecoindRunning from './LeaveMobilecoindRunning';
import LedgerStatus from './LedgerStatus';
import MobilecoindDirectory from './MobilecoindDirectory';
import ResetLedger from './ResetLedger';

const useStyles = makeStyles((theme: Theme) => {
  return {
    cardContainer: {
      paddingBottom: 64,
      paddingTop: 8 * 4,
    },
    modal: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
    },
    paper: {
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    root: {
      backgroundColor: theme.palette.background.dark,
      display: 'flex',
      flexDirection: 'column',
      minHeight: '100vh',
    },
  };
});

const ConfigureMobilecoindView: FC = () => {
  const classes = useStyles();

  return (
    <Container className={classes.cardContainer} maxWidth="sm">
      <Breadcrumbs separator=">" aria-label="breadcrumb">
        <Link
          color="inherit"
          to={routePaths.APP_SETTINGS}
          component={RouterLink}
        >
          <Typography color="textSecondary">Settings</Typography>
        </Link>
        <Typography color="textPrimary">Configure MobileCoinD</Typography>
      </Breadcrumbs>
      <Box
        alignItems="center"
        display="flex"
        justifyContent="space-between"
        my={3}
        flexDirection="column"
      >
        <Typography variant="body2" display="inline" color="textSecondary">
          This panel allows you to customize the behavior of MobileCoinD as well
          as perform some basic resets.
        </Typography>
        <Box py={1} />
        <Typography variant="body2" color="textSecondary">
          The MobileCoin Daemon, or MobileCoinD, is a standalone executable
          which provides blockchain synchronization and wallet services.
        </Typography>
        <Box py={1} />
        <Typography variant="body2" color="textSecondary">
          It creates encrypted, attested connections to validator nodes who are
          participating in federated voting in order to get the current block
          height, block headers, and to submit transactions. These validator
          nodes are considered highly trusted due to their use of SGX, and they
          are used as a reliable source for block information as well as to
          validate and process the proposed transactions from MobileCoinD.
        </Typography>
      </Box>
      <LedgerStatus />
      <LeaveMobilecoindRunning />
      <ResetLedger />
      <MobilecoindDirectory />
    </Container>
  );
};

export default ConfigureMobilecoindView;
