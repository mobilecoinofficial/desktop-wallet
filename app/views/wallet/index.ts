export { default as ChangePasswordView } from './ChangePasswordView';
export { default as ChangePinView } from './ChangePinView';
export { default as ConfigureFullServiceView } from './ConfigureFullServiceView';
export { default as ContactsBookView } from './ContactsBookView';
export { default as DashboardView } from './DashboardView';
export { default as GiftingView } from './GiftingView';
export { default as HistoryView } from './HistoryView';
export { default as PrivacyPolicyView } from './PrivacyPolicyView';
export { default as RetrieveEntropyView } from './RetrieveEntropyView';
export { default as SettingsView } from './SettingsView';
export { default as TermsOfUseView } from './TermsOfUseView';
export { default as TransactionView } from './TransactionView';
export { default as TransactionDetailsView } from './TransactionDetailsView';
