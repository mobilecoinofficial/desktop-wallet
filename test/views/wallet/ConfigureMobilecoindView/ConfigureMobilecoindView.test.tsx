import React from 'react';

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ConfigureMobilecoindView } from '../../../../app/views/wallet';
import renderSnapshot from '../../../renderSnapshot';

describe('ConfigureMobilecoindView', () => {
  describe('component', () => {
    describe('render', () => {
      test('ConfigureMobilecoindView renders correctly', () => {
        const { asFragment } = renderSnapshot(<ConfigureMobilecoindView />);
        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('breadcrumb navigation', () => {
      test('Settings breadcrumb navigates away from current view', () => {
        renderSnapshot(<ConfigureMobilecoindView />);
        const configPanel = screen.queryByText(/This panel allows you to customize the behavior of MobileCoinD/i);
        expect(configPanel).toBeInTheDocument();
        userEvent.click(screen.getByText('Settings'));
        expect(configPanel).not.toBeInTheDocument();
      });
    });
  });
});
