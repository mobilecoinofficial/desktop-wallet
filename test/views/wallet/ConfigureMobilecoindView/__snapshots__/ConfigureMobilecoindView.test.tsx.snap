// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureMobilecoindView component render ConfigureMobilecoindView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-cardContainer MuiContainer-maxWidthSm"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <ol
        class="MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            href="/app/settings"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Settings
            </p>
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          &gt;
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            Configure MobileCoinD
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="MuiBox-root MuiBox-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayInline"
      >
        This panel allows you to customize the behavior of MobileCoinD as well as perform some basic resets.
      </p>
      <div
        class="MuiBox-root MuiBox-root"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        The MobileCoin Daemon, or MobileCoinD, is a standalone executable which provides blockchain synchronization and wallet services.
      </p>
      <div
        class="MuiBox-root MuiBox-root"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        It creates encrypted, attested connections to validator nodes who are participating in federated voting in order to get the current block height, block headers, and to submit transactions. These validator nodes are considered highly trusted due to their use of SGX, and they are used as a reliable source for block information as well as to validate and process the proposed transactions from MobileCoinD.
      </p>
    </div>
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Ledger Status
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayInline"
        >
          The ledger is syncing. This may take awhile.
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <div
          class="MuiTableContainer-root"
        >
          <div
            class="MuiPaper-root MuiTableHead-root MuiPaper-elevation1 MuiPaper-rounded"
            role="rowgroup"
          />
          <table
            aria-label="block status"
            class="MuiTable-root"
          >
            <tbody>
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Block Type
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Current Height
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Maximum Height
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
                  scope="col"
                >
                  % Synced
                </th>
              </tr>
            </tbody>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                  role="cell"
                  scope="row"
                >
                  Network Blocks
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                />
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                  role="cell"
                  scope="row"
                >
                  Local Blocks
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                  role="cell"
                  scope="row"
                >
                  Monitor Blocks
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Network Blocks represent the total blocks within the MobileCoin network. This is the value your Local and Monitor Blocks needs to hit to be 100% synced.
      </p>
      <div
        class="MuiBox-root MuiBox-root"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Local Blocks shows the blocks your mobilecoind has synced the ledger.
      </p>
      <div
        class="MuiBox-root MuiBox-root"
      />
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Monitor Blocks shows the blocks that a specific monitor has synced. Each account has its own Monitor Blocks count. Once this value is synced, you will immediately see the effects of transactions.
      </p>
    </div>
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            MobileCoinD Background Process
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <div
          class="MuiBox-root MuiBox-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            Leave MobileCoinD Active is off
          </p>
          <div
            class="MuiBox-root MuiBox-root"
          >
            <span
              class="MuiSwitch-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input"
                    name="checkedC"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track"
              />
            </span>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayInline"
        >
          MobileCoinD will not continue to sync when you exit the wallet.
        </p>
        <div
          class="MuiBox-root MuiBox-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          By default, MobileCoinD terminates when you exit the wallet. This setting allows you to leave MobileCoinD syncing as a background process. It is best to keep the ledger synced to avoid delays in transactions or balance fetches.
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            Reset Your Ledger
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          In rare cases, your local database may become corrupted and permanently out-of-sync with the ledger. If this occurs, please manually reset your ledger.
        </p>
        <div
          class="MuiBox-root MuiBox-root makeStyles-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            data-testid="submit-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Reset Ledger
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root"
    >
      <div
        class="MuiBox-root MuiBox-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
          >
            MobileCoinD Directory
          </p>
        </legend>
      </div>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <div
          class="MuiBox-root MuiBox-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          Ledger Database Path:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          undefined
        </p>
        <div
          class="MuiBox-root MuiBox-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
        >
          MobileCoinD Database Path:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          undefined
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
